File: test/binaries/24_upward_dylib/hello_world.c
#if LIB_A
#include <stdio.h>
extern void from_b(void);
void from_a(void) { fprintf(stderr, "Hello, World!\n"); }
void entry(void) { from_b(); }
#elif LIB_B
extern void from_a(void);
void from_b(void) { from_a(); }
#else
void entry(void);
int main(void) { entry(); }
#endif

File: test/binaries/34_objc_initialize_cross_dylib/hello_world.m
#import <Foundation/Foundation.h>
#include <stdio.h>

#if LIB_A
@interface A : NSObject @end
@implementation A
+ (void)initialize { fprintf(stderr, "Hello, World!\n"); }
+ (void)touch {}
@end
#elif LIB_B
@interface A : NSObject + (void)touch; @end
@interface B : NSObject @end
@implementation B
+ (void)load { [A touch]; }
@end
void b_anchor(void) {}
#else
void b_anchor(void);
int main(void) { b_anchor(); }
#endif
